lib/token-request.js

```diff
@@ -544,4 +551,20 @@ TokenRequest.prototype.getTokenWithCertificate = function(certificate, thumbprin
   });
 };
 
+TokenRequest.prototype.getTokenWithDeviceCode = function(userCodeInfo, callback) {
+   this._log.info('Getting a token via device code');
+
+   var oauthParameters = this._createOAuthParameters(OAuth2GrantType.DEVICE_CODE);
+   oauthParameters[OAuth2Parameters.CODE] = userCodeInfo[OAuth2Parameters.DEVICE_CODE];
+
+   var interval = userCodeInfo[constants.OAuth2.DeviceCodeResponseParameters.INTERVAL];
```

Comment:
Instead of typing out the namespaces, at the top of the file you can add

```
var OAuth2DeviceCodeResponseParameters = constants.OAuth2.DeviceCodeResponseParaters;
```

to match the other hashes imported from constants.js